contracts/chain-adapters/Rerouter_Adapter.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { AdapterInterface } from "./interfaces/AdapterInterface.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to a target on an arbitrary layer via re-routing messages
+ * through intermediate contracts.
+ * @dev Since this adapter is normally called by the hub pool, the target of both `relayMessage` and `relayTokens`
+ * will be the remote spoke pool due to the constraints of `setCrossChainContracts` outlined in UMIP 157. However, this
+ * contract cannot send anything directly to the this target, since it does not exist on L2. Instead, it "re-routes"
+ * messages to the remote network via intermediate forwarder contracts,beginning with an L2 forwarder, which is set as the
+ * `l2Target` in this contract. Each forwarder contract contains logic which determines the path a message or token relay
+ * must take to ultimately arrive at the spoke pool. There should be one of these adapters for each L3 spoke pool deployment.
+ * @dev All forwarder contracts, including `l2Target`, must implement ForwarderBase in order for tokens and messages to be
+ * automatically relayed to the subsequent layers.
+ * @dev Public functions calling external contracts do not guard against reentrancy because they are expected to be
+ * called via delegatecall, which will execute this contract's logic within the context of the originating contract.
+ * For example, the HubPool will delegatecall these functions, therefore its only necessary that the HubPool's methods
+ * that call this contract's logic guard against reentrancy.
+ * @custom:security-contact bugs@across.to
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Rerouter_Adapter is AdapterInterface {
+    address public immutable l1Adapter;
```

Comment:
nit: immutable and constant variables should be in ALL_CAPS

Reply:
https://github.com/across-protocol/contracts/pull/607/commits/1274920a219caf9f9fc492f6dac6e067a979c67d